src/components/molecules/MainContainer/MainContainer.tsx

```diff
@@ -0,0 +1,16 @@
+import styles from "./MainContainer.module.scss";
+import { clsx } from "clsx";
+import { ReactNode } from "react";
+
+interface MainLayoutProps {
```

Comment:
I think this could be handled as a HOC (high-order component) and define the style in the passed component.
[Docu](https://www.geeksforgeeks.org/how-to-use-higher-order-components-in-react/)